connectors/hub-concur/src/main/java/com/vmware/connectors/concur/service/HubConcurService.java

```diff
@@ -0,0 +1,232 @@
+package com.vmware.connectors.concur.service;
+
+import com.vmware.connectors.common.payloads.response.Card;
+import com.vmware.connectors.common.payloads.response.CardAction;
+import com.vmware.connectors.common.payloads.response.CardActionInputField;
+import com.vmware.connectors.common.payloads.response.CardActionKey;
+import com.vmware.connectors.common.payloads.response.CardBody;
+import com.vmware.connectors.common.payloads.response.CardBodyField;
+import com.vmware.connectors.common.payloads.response.CardBodyFieldType;
+import com.vmware.connectors.common.payloads.response.Cards;
+import com.vmware.connectors.common.utils.CardTextAccessor;
+import com.vmware.connectors.common.utils.CommonUtils;
+import com.vmware.connectors.concur.domain.ExpenseReportResponse;
+import com.vmware.connectors.concur.domain.PendingApprovalResponse;
+import com.vmware.connectors.concur.util.HubConcurUtil;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.HtmlUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.MediaType.APPLICATION_XML;
+
+@Service
+public class HubConcurService {
+
+    private static final Logger logger = LoggerFactory.getLogger(HubConcurService.class);
+
+    private final WebClient rest;
+    private final CardTextAccessor cardTextAccessor;
+    private final Resource concurRequestTemplate;
+    private final String serviceAccountAuthHeader;
+
+    @Autowired
+    public HubConcurService(
+            WebClient rest,
+            CardTextAccessor cardTextAccessor,
+            @Value("classpath:static/templates/concur-request-template.xml") Resource concurRequestTemplate,
+            @Value("${concur.service-account-auth-header}") String serviceAccountAuthHeader
```

Comment:
I am still doubtful, how they will be able to pass the user-specific service token from ws1 hub.

Reply:
They cannot.  That is why they must have the service account.  Since Concur's APIs don't really allow us to use oauth (rememeber the same problem you had in the non-hub concur connector where we had to use basic auth to get the oauth token).

For hub, however, we'll  never get basic auth credentials, so they have to use a service account and implement filtering and privilege checks themselves.